cupy/core/_routines_indexing.pyx

```diff
@@ -87,22 +89,36 @@ cpdef ndarray _ndarray_argwhere(ndarray self):
             scan_dtype = numpy.int32
         else:
             scan_dtype = numpy_int64
-        scan_index = _math.scan(nonzero, op=_math.scan_op.SCAN_SUM,
-                                dtype=scan_dtype)
+
+        chunk_size = 512
+        if nonzero.size > chunk_size:
+            # TODO(anaruse): We need to set an appropriate threshold, as
```

Comment:
Maybe we can use cupyx.optimize to have optuna calculate this parameter?

Reply:
Let's update this comment and it will be good to go

Reply:
Sorry for the delay. I've updated the comment.